axlearn/cloud/gcp/job.py

```diff
@@ -639,6 +639,327 @@ def _execute(self) -> Any:
         )
 
 
+class GPUGKEJob(GKEJob):
+    """A GPU job represented as a k8s JobSet.
+
+    See also `gke_runner` as an example.
+    """
+
+    @config_class
+    class Config(GKEJob.Config):
+        """Configures GPUGKEJob.
+
+        Attributes:
+            accelerator: GPU configuration.
+        """
+
+        accelerator: AcceleratorConfig = AcceleratorConfig()
+
+    @classmethod
+    def define_flags(cls, fv: flags.FlagValues):
+        super().define_flags(fv)
+        common_kwargs = dict(flag_values=fv, allow_override=True)
+        accelerator_flags(**common_kwargs)
+
+    @classmethod
+    def from_flags(cls, fv: flags.FlagValues, **kwargs) -> Config:
+        cfg: GPUGKEJob.Config = super().from_flags(fv, **kwargs)
+        cfg.accelerator.set(instance_type=fv.instance_type, num_replicas=fv.num_replicas)
+        return cfg
+
+    def __init__(self, cfg: Config):
+        bundler_cfg = cfg.bundler
+        bundler_cfg = getattr(bundler_cfg, "inner", bundler_cfg)
+        if bundler_cfg is None or not issubclass(bundler_cfg.klass, BaseDockerBundler):
+            raise NotImplementedError(f"Only docker bundler supported, got: {bundler_cfg}")
+        super().__init__(cfg)
+
+    def _build_sidecar_container(self) -> Nested[Any]:
+        cfg: GPUGKEJob.Config = self.config
+
+        if cfg.accelerator.instance_type.startswith("a3-highgpu"):
+            volume_mounts = [
```

Comment:
Add a reference/comment on what this sidecar is doing and why we only do this for a3-highgpu?

Reply:
done